src/thermal.cpp

```diff
@@ -117,6 +117,18 @@ ThermalScattering::ThermalScattering(
         }
       }
       if (!found) {
+        // If no pairs found, check if the desired temperature falls within
+        // bounds' tolerance
+        if (std::abs(T - temps_available[0]) <=
+            settings::temperature_tolerance) {
+          temps_to_read.push_back(std::round(temps_available[0]));
+          break;
+        }
+        if (std::abs(T - temps_available[n - 1]) <=
+            settings::temperature_tolerance) {
+          temps_to_read.push_back(std::round(temps_available[n - 1]));
```

Comment:
> Thanks for this improvement @joshmay1! Sorry it took me a while to review this in full. I have a few small comments but overall it looks like it's in really good shape.

You're welcome @paulromano and thank you for the review. Sorry for the goofy comparisons, but those should be fixed now. One thing I noticed is that there doesn't seem to be an explicit check if a thermal scattering library has already been pushed back as there is with the nuclide temperatures. Was there a reason for this?

Reply:
I don't think so -- that looks like a bug waiting to happen! Do you want to go ahead and fix that on this brach?